### ISSUE TITLE
Exporting onto the Source file exports a full-white video

### ISSUE THREAD

**Issue by Ekleog** [mltframework/shotcut#840]
Steps to reproduce:
- `shotcut foo.mp4`
- Add a crop filter
- Export to `foo.mp4`
- Observe that the exported file is pure white

Suggested solution: Either have export work properly (presumably hard), or refuse to export to the source file (presumably easier).

(I guess this'll be closed like #839, but wanted to have the fact I believe this is a bug written down somewhere in the bug tracker in case other people ever read it, so they can have an idea of how to make shotcut more new-user-friendly)

**Comment by ddennedy** [comment]
Actually, we already have code to prevent this, but it is not working in this case or set of steps. I will try to reproduce it.

**Comment by ddennedy** [comment]
I do not reproduce this with version 20.02 on Linux. Step 1 fails because you must qualify the file name with `"$PWD/foo.mp4"` or use some other path. Next, in step 3 when I enter test.mp4 (I am using test.mp4 as source), I get this

![Screenshot from 2020-03-21 18-11-13](https://user-images.githubusercontent.com/1146683/77240117-78dddf00-6b9f-11ea-8631-555cac36adbc.png)

If I choose Yes, I get the warning dialog from the code I mentioned:

![Screenshot from 2020-03-21 18-12-31](https://user-images.githubusercontent.com/1146683/77240121-8d21dc00-6b9f-11ea-8746-326e4602a62a.png)

Even when I leave off ".mp4" in the file save dialog, to let the automatic extension be added, I do not reproduce the problem. Even if I add the clip from Source to Timeline and export from Timeline, I do not reproduce it. This report lacks version and OS information and perhaps some other steps.

**Comment by ddennedy** [comment]
I did reproduce this on the flatpak. I also found the first step works with a relative path with the flatpak. I was testing in previous with the portable and snap. I made a fix for that loading issue. Now, as for the export problem with flatpak, I am very surprised to see that code fail. 
I still get a warning:

![Screenshot from 2020-03-21 18-41-03](https://user-images.githubusercontent.com/1146683/77240498-92812580-6ba3-11ea-91b5-7e4893359e86.png)

Oh, I see now from looking at the export job's XML. The reason it failed is because the target file name, which is absolute, is not found in the XML, which has only a relative path. Typically, this is prevented because the app launch directory is different than the current directory causing a full path in the XML. I also reproduced this by running my development executable.
